TodoList/TodoList.Api/Controllers/NodesController.cs

```diff
@@ -1,44 +1,49 @@
 ﻿using System;
-using System.Collections.Generic;
-using System.Linq;
 using System.Net;
 using System.Threading.Tasks;
 using System.Web.Http;
-using TodoList.Api.Models;
+using TodoList.Contracts.Api;
+using TodoList.Contracts.DAL;
 
 namespace TodoList.Api.Controllers
 {
-    [Route("api/v1/Nodes/{id:guid?}", Name = "Nodes")]
+    [Route("api/v1/nodes/{id:guid?}", Name = "nodes")]
     public class NodesController : ApiController
     {
-        private NodeModel[] Nodes { get; }
+        private readonly INodesRepository _repository;
+        private readonly ILocationHelper _locationHelper;
 
-        public NodesController()
+        public NodesController() { }
+
+        public NodesController(INodesRepository repository, ILocationHelper locationHelper)
         {
-            Nodes = InitializeData().ToArray();
+            _repository = repository;
+            _locationHelper = locationHelper;
         }
 
         public async Task<IHttpActionResult> GetAsync()
-            => await Task.FromResult<IHttpActionResult>(Ok(Nodes));
+            => await Task.FromResult<IHttpActionResult>(Ok(_repository.GetAllAsync().Result));
 
         public async Task<IHttpActionResult> GetAsync(Guid id)
-            => await Task.FromResult<IHttpActionResult>(Ok(Nodes[0]));
+            => await Task.FromResult<IHttpActionResult>(Ok(_repository
+                .GetByIdAsync(new Guid("d237bdda-e6d4-4e46-92db-1a7a0aeb9a72")).Result));
```

Comment:
Dummy logic should be moved to repository, this should pretend the repository is real